diff --git a/Include/RmlUi/Core/Colour.h b/Include/RmlUi/Core/Colour.h
index d63540a8..3b25d881 100644
--- a/Include/RmlUi/Core/Colour.h
+++ b/Include/RmlUi/Core/Colour.h
@@ -80,3 +80,3 @@
-			ColourType((red * alpha) / 255),
-			ColourType((green * alpha) / 255),
-			ColourType((blue * alpha) / 255),
+			ColourType((red * alpha + 127) / 255),
+			ColourType((green * alpha + 127) / 255),
+			ColourType((blue * alpha + 127) / 255),
@@ -93,4 +93,4 @@
-			ColourType(red * (new_alpha / 255.f)),
-			ColourType(green * (new_alpha / 255.f)),
-			ColourType(blue * (new_alpha / 255.f)),
-			ColourType(new_alpha),
+			ColourType(red * (new_alpha / 255.f) + 0.5f),
+			ColourType(green * (new_alpha / 255.f) + 0.5f),
+			ColourType(blue * (new_alpha / 255.f) + 0.5f),
+			ColourType(new_alpha + 0.5f),
